mundane-bytes/src/main/scala/com/ambiata/mundane/bytes/Buffer.scala

```diff
@@ -0,0 +1,26 @@
+package com.ambiata.mundane.bytes
+
+case class Buffer(bytes: Array[Byte], offset: Int, var length: Int) {
```

Comment:
I think a big comment is the go, specifically it would be good to explain the background, the reasons why certain things are immutable or not, the expected use etc...